### PR TITLE
documented undocumented endpoints

### PR DESCRIPTION
**Pull request by anwilkie**
# Motivation and Context
This pull request provides documentation for endpoints that were missing documentation.

# What has changed
Updated API.md to include the missing endpoints, as well as making note of a redundant endpoint.

# Links
[Trello card](https://trello.com/c/OC8gdr83)

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -152,3 +152,41 @@ The payload should be a classifier object, with a classifier type selector `name
...
+    "LongName":"test-long-name"
+}
+```
+
+An `HTTP 500 Internal Server Error` status code is returned if the PUT request was unsuccessful.
+
+## Get Legal Bases
+* `GET /legal-bases` returns a list of legal bases (?).
+
+NOTE: Although this endpoint is defined, there exists no code for it anywhere. As such, it is redundant.
```
Comment (gavinleeedwards) [inline API.md]: There's code for this endpoint, as it calls function AllLegalBases.  See: https://github.com/ONSdigital/rm-survey-service/blob/cb968283ad840ac3ba8a4a19b8bbf422c44914af/models/surveys.go#L573  It's used by response-operations-ui to populate the list of legal bases when creating a new survey.

File: API.md [hunk 2]
```diff
@@ -152,3 +152,41 @@ The payload should be a classifier object, with a classifier type selector `name
 An `HTTP 404 Not Found` status code is returned if the survey with the specified ID could not be found.
 
 An `HTTP 409 Conflict` status code is returned if a classifier type selector already exists for any of the names in the payload.
+
+## Post New Survey
+* `POST /surveys` will create a new survey.
+
+The payload should be a byte array, with a `ShortName`, a `LongName`, a `SurveyRef`, a `LegalBasis`, and a `SurveyType`.
```
Comment (jonnyshaw89) [inline API.md]: From looking through the code shouldn't the payload match this struct?
```
// Survey represents the details of a survey.
type Survey struct {
	ID            string `json:"id"`
	ShortName     string `json:"shortName" validate:"required,no-spaces,max=20"`
	LongName      string `json:"longName" validate:"required,max=100"`
	Reference     string `json:"surveyRef" validate:"required,max=20"`
	LegalBasis    string `json:"legalBasis"`
	SurveyType    string `json:"surveyType"`
	LegalBasisRef string `json:"legalBasisRef"`
}
```

Comment (jonnyshaw89) [reply API.md]: The same struct is also used for the request below.

Comment (necrophonic) [reply API.md]: The payload should be a `json` doc (the fields of which match those in the struct)

File: API.md [hunk 3]
```diff
@@ -152,3 +152,63 @@ The payload should be a classifier object, with a classifier type selector `name
...
+
+## Post New Survey
+* `POST /surveys` will create a new survey.
+
+The payload should be a JSON document, with an `ID`, a `ShortName`, a `LongName`, a `Reference`, a `LegalBasis`, a `SurveyType` as strings, and `Classifiers` as a list.
+
+### Example JSON payload
+```json
+{
+    "ID": "efa868fb-fb80-44c7-9f33-d6800a17c4da",
```
Comment (jonnyshaw89) [inline API.md]: these keys are incorrect. with a Golang struct the bit after `json:` is the key name
`ID            string json:"id"`
so for example this should be `id` not `ID`

File: API.md [hunk 4]
```diff
@@ -152,3 +152,63 @@ The payload should be a classifier object, with a classifier type selector `name
...
+```json
+{
+    "id": "efa868fb-fb80-44c7-9f33-d6800a17c4da",
+    "shortName": "test-short-name", 
+    "longName": "test-long-name",
+    "surveyRef": "456",
+    "legalBasis": "Statistics of Trade Act 1947",
+    "surveyType": "Social",
+    "legalBasisRef": "STA1947",
+    "classifiers,omitempty": [
```
Comment (jonnyshaw89) [inline API.md]: `omitempty` is an annotation for the golang json marshaller which means to not create that entity of it is not present. essentially stating its optionality.
Basically, this doesn't form part of the payload. It just highlights that its optional

**Pull request merged**